components/ide-service/pkg/server/server.go

```diff
@@ -60,8 +68,38 @@ func Start(logger *logrus.Entry, cfg *config.ServiceConfiguration) error {
 	if err != nil {
 		return fmt.Errorf("failed to initialize ide-service: %w", err)
 	}
+	var (
+		dockerCfg   *configfile.ConfigFile
+		dockerCfgMu sync.RWMutex
+	)
+
+	resolverProvider := func() remotes.Resolver {
+		var resolverOpts docker.ResolverOptions
+
+		dockerCfgMu.RLock()
+		defer dockerCfgMu.RUnlock()
+		if dockerCfg != nil {
+			resolverOpts.Hosts = docker.ConfigureDefaultRegistries(
+				docker.WithAuthorizer(authorizerFromDockerConfig(dockerCfg)),
+			)
+		}
+
+		return docker.NewResolver(resolverOpts)
+	}
+	if cfg.DockerCfg != "" {
+		dockerCfg = loadDockerCfg(cfg.DockerCfg)
+		err = watch.File(ctx, cfg.DockerCfg, func() {
```

Comment:
@nandajavarma Is it good? I've seen some discussions today that we need to watch folder not a file? How do you do it for registry-facade?